cmd/trade.go

```diff
@@ -722,6 +726,26 @@ func runTradeCmd(options inputs) {
 	bot.Start()
 }
 
+func getUserID(l logger.Logger, botConfig trader.BotConfig) (string, error) {
+	var userIDPrehash string
+	if botConfig.IsTradingSdex() {
+		userIDPrehash = botConfig.TradingAccount()
+	} else {
+		exchangeAPIKeys := botConfig.ExchangeAPIKeys.ToExchangeAPIKeys()
+		userIDPrehash = exchangeAPIKeys[0].Key
+	}
+
+	// hash avoids exposing the user account or api key
+	userIDHashed, e := utils.HashString(userIDPrehash)
+	if e != nil {
+		logger.Fatal(l, fmt.Errorf("could not create user id: %s", e))
```

Comment:
wherever we can return an error we should. it's better than deciding how to handle the error at these inner function levels which is an unexpected "side effect" of the function.